### PR TITLE
Feat: Add Identicon Widget

### PR DESCRIPTION
**Pull request by parodyBit**
add lib/widgets/identicon.dart
roll dart version in pubspec.yaml to 2.14.0 to use the triple shift operator `>>>` add IdenticonWidget for the wallet id image.

### CODE REVIEW

File: lib/widgets/identicon.dart [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+import 'package:fixnum/fixnum.dart';
+
+/// Identicon widget with customized painter
+///
+/// [size] - number of blocks on x and y axis
+/// [color] - main color of the identicon (43% probability)
+/// [bgColor] - background color of the identicon (43% probability)
+/// [spotColor] - additional color of the identicon (13% probability)
+///
+class IdenticonWidget extends StatelessWidget {
```
Comment (gabaldon) [inline lib/widgets/identicon.dart:13]: I think we can delete the widget word when naming a widget
```suggestion
class Identiconextends StatelessWidget {
```

File: lib/widgets/layouts/layout.dart [hunk 2]
```diff
@@ -35,13 +36,16 @@ class LayoutState extends State<Layout> with TickerProviderStateMixin {
...
+    String walletId = Locator.instance.get<ApiDatabase>()
+        .walletStorage.currentWallet.id;
     return MouseRegion(
         cursor: SystemMouseCursors.click,
         child: GestureDetector(
           child: Container(
             color: WitnetPallet.white,
             width: 30,
             height: 30,
+            child: IdenticonWidget(seed: walletId ?? '', size: 8),
```
Comment (gabaldon) [inline lib/widgets/layouts/layout.dart:48]: The walletId can't be null
```suggestion
            child: IdenticonWidget(seed: walletId , size: 8),
```

**Review by gabaldon — CHANGES_REQUESTED** [review]

File: lib/widgets/identicon.dart (hunk 1 above)
Comment (parodyBit) [reply lib/widgets/identicon.dart:13]: makes sense, i will change it

File: pubspec.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ publish_to: 'none'
 version: 0.0.1-dev
 
 environment:
-  sdk: ">=2.12.0 <3.0.0"
```
[the remark this answers is not in the document]
Comment (parodyBit) [reply pubspec.yaml:9]: `>>> 0` ensures we have got an integer between 0 and 0xFFFFFFFF

**Review by gabaldon — APPROVED** [review]

**Pull request merged**